test/e2e-common.sh

```diff
@@ -127,3 +175,58 @@ function dump_extra_cluster_state() {
     done
   done
 }
+
+function install_istio {
+  if [[ -z "${ISTIO_VERSION}" ]]; then
+    readonly ISTIO_VERSION="1.4.3"
+  fi
+  echo ">> Installing Istio: ${ISTIO_VERSION}"
+
+  local istio_base="./third_party/istio-${ISTIO_VERSION}"
+  INSTALL_ISTIO_CRD_YAML="${istio_base}/istio-crds.yaml"
+  INSTALL_ISTIO_YAML="${istio_base}/istio-minimal.yaml"
+
+  echo "Istio CRD YAML: ${INSTALL_ISTIO_CRD_YAML}"
+  echo "Istio YAML: ${INSTALL_ISTIO_YAML}"
+
+  echo ">> Bringing up Istio"
+  echo ">> Running Istio CRD installer"
+  kubectl apply -f "${INSTALL_ISTIO_CRD_YAML}" || return 1
+  wait_until_batch_job_complete istio-system || return 1
+  UNINSTALL_LIST+=( "${INSTALL_ISTIO_CRD_YAML}" )
+
+  echo ">> Running Istio"
+  kubectl apply -f "${INSTALL_ISTIO_YAML}" || return 1
+  UNINSTALL_LIST+=( "${INSTALL_ISTIO_YAML}" )
+}
+
+# Installs Knative Serving in the current cluster, and waits for it to be ready.
+function install_knative_serving {
```

Comment:
testing eventing should not need to depend on serving.

Reply:
I think there should be tests that tests eventing & serving interoperation. Maybe you're right that this is not a place for that. I don't know. 

I proposed to check that against serving as I fought that this might be a breaking point for upgrade tests.

Reply:
There should be tests that do that but I think we should decouple those two things.

Reply:
Where we should put tests for integration of eventing & serving?